hack/test_errata.py

```diff
@@ -0,0 +1,598 @@
+import datetime
+import errata
+import github
+import http
+import json
+import os
+import re
+import tempfile
+import unittest
+import urllib
+from collections import namedtuple
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+def test_extract_errata_number_from_body_valid_url():
+    body = "errata is https://errata.devel.redhat.com/advisory/12345"
+    assert errata.extract_errata_number_from_body(body) == 12345
+
+    body = "errata is https://errata.devel.redhat.com/advisory/9999"
+    assert errata.extract_errata_number_from_body(body) == 9999
+
+    body = "errata is https://errata.devel.redhat.com/advisory/"
+    assert errata.extract_errata_number_from_body(body) == None
+
+    body = "errata is https://errata.devel.redhat.com/advisory/invalid"
+    assert errata.extract_errata_number_from_body(body) == None
+
+
+def test_extract_errata_number_from_body_invalid_url():
+    # Possible invalid body
+    # body = "errata is https://errata.devel.redhat.com/advisory/advisory/12345"
+    # assert errata.extract_errata_number_from_body(body) == None
+
+    body = "errata is https://errata.devel.redhat.com/"
+    assert errata.extract_errata_number_from_body(body) == None
+
+
+def test_load():
+    with tempfile.TemporaryDirectory() as tempdir:
+        cachepath = os.path.join(tempdir, "cache.json")
+        cache = {"foo": "bar"}
+
+        with open(cachepath, 'w') as file:
+            json.dump(cache, file)
```

Comment:
I think it's probably better to test `save` and `load` as a pair, instead of inlining a `save` implementation here so you can test `load` independently.  Something like:

1. Create a temp dir
2. Try to load `cachepath`.  Confirm you get `{}`.
3. Add some dummy data.
4. `save`
5. `load`
6. Confirm you got the dummy data back unchanged.